src/_P092_DLbus.ino

```diff
@@ -0,0 +1,1180 @@
+      DataSettings.MaxDoublePulseWidth = P092_double_max_width_488;
+
+      DataSettings.Deviceuint8_t0 = 0x90;
+      DataSettings.Deviceuint8_t1 = 0x9F;
+      DataSettings.Deviceuint8_ts = 2;
+      DataSettings.DontCareuint8_ts = 1;
+      DataSettings.TimeStampuint8_ts = 5;
+      DataSettings.MaxSensors = 6;
+      DataSettings.MaxExtSensors = 9;
+      DataSettings.Outputuint8_ts = 1;
+      DataSettings.Speeduint8_ts = 1;
+      DataSettings.Analoguint8_ts = 2;
+      DataSettings.Volumeuint8_ts = 0;
+      DataSettings.MaxHeatMeters = 3;
+      DataSettings.CurrentHmuint8_ts = 2;
+      DataSettings.MWhuint8_ts = 2;
+      DataSettings.IdxCRC = 61;
+
+      P092_OptionValueDecimals[6] = 1; //[0,1kW]     F("Heat power (kW)")
+      break;
+  }
+  DataSettings.IdxSensor               = DataSettings.Deviceuint8_ts + DataSettings.DontCareuint8_ts + DataSettings.TimeStampuint8_ts;
+  DataSettings.IdxExtSensor            = DataSettings.IdxSensor + 2 * DataSettings.MaxSensors;
+  DataSettings.IdxOutput               = DataSettings.IdxExtSensor + 2 * DataSettings.MaxExtSensors;
+  DataSettings.IdxDrehzahl             = DataSettings.IdxOutput + DataSettings.Outputuint8_ts;
+  DataSettings.IdxAnalog               = DataSettings.IdxDrehzahl + DataSettings.Speeduint8_ts;
+  DataSettings.IdxHmRegister           = DataSettings.IdxAnalog + DataSettings.Analoguint8_ts;
+  DataSettings.IdxVolume               = DataSettings.IdxHmRegister + 1;
+  DataSettings.IdxHeatMeter1           = DataSettings.IdxVolume + DataSettings.Volumeuint8_ts;
+  DataSettings.IdxkWh1                 = DataSettings.IdxHeatMeter1 + DataSettings.CurrentHmuint8_ts;
+  DataSettings.IdxMWh1                 = DataSettings.IdxkWh1 + 2;
+  DataSettings.IdxHeatMeter2           = DataSettings.IdxMWh1 + DataSettings.MWhuint8_ts;
+  DataSettings.IdxkWh2                 = DataSettings.IdxHeatMeter2 + DataSettings.CurrentHmuint8_ts;
+  DataSettings.IdxMWh2                 = DataSettings.IdxkWh2 + 2;
+  DataSettings.IdxHeatMeter3           = DataSettings.IdxMWh2 + DataSettings.MWhuint8_ts;
+  DataSettings.IdxkWh3                 = DataSettings.IdxHeatMeter3 + DataSettings.CurrentHmuint8_ts;
+  DataSettings.IdxMWh3                 = DataSettings.IdxkWh3 + 2;
+}
+/*********************************************************************************************\
+  DLBus subs to get values from the P092_receiving bitstream
+\*********************************************************************************************/
+// one data frame has <DataSettings.Datauint8_ts> data uint8_ts + SYNC, e.g. 64 * (8+1+1) + 16 = 656
+// 656 * 2 = 1312 (twice as much as a data frame is saved
+// so there's one complete data frame
+
+#define MaxDatauint8_ts 64
+#define AdditionalRecuint8_ts 2
+#define MaxDataBits (((MaxDatauint8_ts + AdditionalRecuint8_ts) * (8 + 1 + 1) + 16) * 4)
+volatile uint8_t BitStream[MaxDataBits];           // received bit stream
```

Comment:
All volitile variables are only used in the interrupt function.
It has to be fast, a bit change can occour every 2 milli seconds.
On most of the DL bus devices the bus clock is 488Hz.

Reply:
I forgot one thing.
The time between the bit changes has to be measured quite accurate because the DL bus got a strange protocol:
Two eqal bits in sequence are without bit change, just the double length of the high signal. If there is another high bit following, a short low pulse is issued. The same for lows in sequence.